Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm

```diff
@@ -187,6 +187,15 @@
     setNeedsDisplay(IntRect(IntPoint(), expandedIntSize(m_parameters.size)));
 }
 
+DestinationColorSpace RemoteLayerBackingStore::colorSpace() const
+{
+#if HAVE(IOSURFACE_RGB10)
+    if (m_parameters.type == Type::IOSurface && m_parameters.deepColor)
```

Comment:
Can we have the `#if HAVE(..)` and this condition in a helper function since it is repeated in `RemoteLayerBackingStore::pixelFormat()` also? e.g. `isDeepColorBackingStore()`, `isHalfFloatPixelFormatSupported()`,` isRGB10PixelFormatSupported()` or something else.

Reply:
An alternative probably-even-better approach is to adopt `PixelBufferFormat` (and maybe even thread it through ImageBuffer::create too)

Reply:
(and then we only have one function again)